### ISSUE TITLE
dnn: Unknown layer type Shape in op dropout/Shape in function populateNet

### ISSUE THREAD

**Issue by Ape** [opencv/opencv#9563]
##### System information (version)
- OpenCV => 3.3.0
- Operating System / Platform => Arch Linux 64-bit
- Compiler => g++ 7.1.1

##### Detailed description
I am trying to load a tensorflow model in OpenCV, but it fails on `populateNet` with:

```
dropout/Shape:Shape(Relu_2)
T:0
out_type:[ ]
OpenCV Error: Unspecified error (Unknown layer type Shape in op dropout/Shape) in populateNet, file /build/opencv/src/opencv-3.3.0/modules/dnn/src/tensorflow/tf_importer.cpp, line 883
terminate called after throwing an instance of 'cv::Exception'
  what():  /build/opencv/src/opencv-3.3.0/modules/dnn/src/tensorflow/tf_importer.cpp:883: error: (-2) Unknown layer type Shape in op dropout/Shape in function populateNet
```

I think this is related to https://github.com/opencv/opencv_contrib/issues/1241

Are there any plans for supporting `shape` layers?

**Comment by dkurt** [comment]
@Ape, it's a problem of graph representation. Dropout is used only for training. There is a flag to switch dropout ops to evaluation mode but it doesn't remove the related ops from the graph. The best solution for me - freeze a graph using weights of a whole network (dropout has no weights) but graph definition without them. See https://github.com/opencv/opencv/issues/9536.

**Comment by Ape** [comment]
I confirm this is a duplicate of #9536.
